R/ui.R

```diff
@@ -79,19 +68,102 @@
+                   "should be set by using parameter .options."),
+            .frequency = "once", .frequency_id = "lambda_warning")
+    }
+    if (hasArg(tau)) {
+        rlang::warn(
+            paste0("Warning: Function parameter tau is deprecated in this new ",
+                   "release of harmony. It will be ignored for this function ",
+                   "call and please remove parameter tau in future ",
+                   "function calls. For advanced users, value of tau ",
+                   "should be set by using parameter .options."),
+            .frequency = "once", .frequency_id = "tau_warning")
+    }
+    if (hasArg(block.size)) {
+        rlang::warn(
+            paste0("Warning: Function parameter block.size is deprecated in this new ",
+                   "release of harmony. It will be ignored for this function ",
+                   "call and please remove parameter block.size in future ",
+                   "function calls. For advanced users, value of block.size ",
+                   "should be set by using parameter .options."),
+            .frequency = "once", .frequency_id = "block.size_warning")
+    }
+    if (hasArg(max.iter.harmony)) {
+        rlang::warn(
+            paste0("Warning: Function parameter max.iter.harmony is replaced ",
+                   "with parameter max_iter in this new release of harmony. ",
+                   "It will be ignored for this function call and please use ",
+                   "parameter max_iter in future function calls."),
+            .frequency = "once", .frequency_id = "max.iter.harmony_warning")
+    }
+    if (hasArg(max.iter.cluster)) {
+        rlang::warn(
+            paste0("Warning: Function parameter max.iter.cluster is deprecated in this new ",
+                   "release of harmony. It will be ignored for this function ",
+                   "call and please remove parameter max.iter.cluster in future ",
+                   "function calls. For advanced users, value of max.iter.cluster ",
+                   "should be set by using parameter .options."),
+            .frequency = "once", .frequency_id = "max.iter.cluster_warning")
+    }
+    if (hasArg(epsilon.cluster)) {
+        rlang::warn(
+            paste0("Warning: Function parameter epsilon.cluster is deprecated in this new ",
+                   "release of harmony. It will be ignored for this function ",
+                   "call and please remove parameter epsilon.cluster in future ",
+                   "function calls. For advanced users, value of epsilon.cluster ",
+                   "should be adjusted by using parameter .options."),
+            .frequency = "once", .frequency_id = "epsilon.cluster_warning")
+    }
+    if (hasArg(epsilon.harmony)) {
+        rlang::warn(
+            paste0("Warning: Function parameter epsilon.harmony is deprecated in this new ",
+                   "release of harmony. It will be ignored for this function ",
```

Comment:
Messages could be clearer. Something like the one below is more concise:

Function parameter `` is deprecated, and its user set value will be ignored. Advanced users can set the `` value by passing harmony_options(``) to the .options function parameter.

Reply:
Sounds good